src/Service/PaymentMethod/PaymentMethodRestrictionValidation/AmountPaymentMethodRestrictionValidator.php

```diff
@@ -0,0 +1,81 @@
+<?php
+/**
+ * Copyright (c) 2012-2020, Mollie B.V.
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * - Redistributions of source code must retain the above copyright notice,
+ *    this list of conditions and the following disclaimer.
+ * - Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE AUTHOR AND CONTRIBUTORS ``AS IS'' AND ANY
+ * EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE AUTHOR OR CONTRIBUTORS BE LIABLE FOR ANY
+ * DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+ * OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH
+ * DAMAGE.
+ *
+ * @author     Mollie B.V. <info@mollie.nl>
+ * @copyright  Mollie B.V.
+ * @license    Berkeley Software Distribution License (BSD-License 2) http://www.opensource.org/licenses/bsd-license.php
+ *
+ * @category   Mollie
+ *
+ * @see       https://www.mollie.nl
+ * @codingStandardsIgnoreStart
+ */
+
+namespace Mollie\Service\PaymentMethod\PaymentMethodRestrictionValidation;
+
+use Mollie\Adapter\LegacyContext;
+use MolPaymentMethod;
+use PrestaShop\Decimal\Number;
+
+/** Validator to check if cart total is valid for amount restrictions */
+class AmountPaymentMethodRestrictionValidator implements PaymentMethodRestrictionValidatorInterface
+{
+    /** @var LegacyContext */
+    private $context;
+
+    public function __construct(
+        LegacyContext $context
+    ) {
+        $this->context = $context;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function isValid(MolPaymentMethod $paymentMethod)
+    {
+        $orderTotal = $this->context->getCart()->getOrderTotal();
+
+        $orderTotalAmount = (new Number((string) $this->context->getCurrency()->getConversionRate()))->times(new Number((string) $orderTotal))->toPrecision(2);
```

Comment:
Calculator service maybe for this, precision is not always 2 (Japanese and other currencies as an example)

Reply:
we are converting any currency that we get to EUR here. Also changed it now but still it would be fine because we take total and do times conversion rate to euro